src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -32,32 +37,26 @@ MasterMetadataServiceImpl ::chunk_server_manager() {
   return server::ChunkServerManager::GetInstance();
 }
 
-grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
-    const protos::grpc::OpenFileRequest* request,
-    protos::grpc::OpenFileReply* reply) {
-  // Step 1. Create file metadata
+grpc::Status MasterMetadataServiceImpl::HandleFileChunkCreation(
+  const protos::grpc::OpenFileRequest* request,
+  protos::grpc::OpenFileReply* reply) {
   const std::string& filename(request->filename());
-  LOG(INFO) << "MasterMetadataService handling file creation: " << filename;
-
-  google::protobuf::util::Status status(
-      metadata_manager()->CreateFileMetadata(filename));
-  if (!status.ok()) {
-    LOG(ERROR) << "File metadata creation failed: " << status.error_message();
-    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
-  }
+  const uint32_t chunk_index(request->chunk_index());
 
-  // Step 2. Create the first file chunk for this file
+  // Step 1. Create the file chunk
   google::protobuf::util::StatusOr<std::string> chunk_handle_or(
-      metadata_manager()->CreateChunkHandle(filename, 0));
+      metadata_manager()->CreateChunkHandle(filename, chunk_index));
   if (!chunk_handle_or.ok()) {
-    LOG(ERROR) << "Chunk handle creation failed: " 
-               << chunk_handle_or.status().error_message();
+    LOG(ERROR) << "Chunk handle creation failed: " << chunk_handle_or.status();
     return common::utils::ConvertProtobufStatusToGrpcStatus(
                chunk_handle_or.status());
+  } else {
+    LOG(INFO) << "Chunk handle created: " << chunk_handle_or.ValueOrDie()
+              << " for file " << filename;
   }
   const std::string& chunk_handle(chunk_handle_or.ValueOrDie());
 
-  // Step 3. Allocate chunk servers
+  // Step 2. Allocate chunk servers for this file chunk
```

Comment:
your comment on line 66 "prepare the InitChunkFileReply" seems incorrect, as you are modifying `OpenFileReply`

Reply:
Will correct it.